app/controllers/spree/users_controller.rb

```diff
@@ -12,9 +12,14 @@ class UsersController < Spree::StoreController
     before_action :set_locale
     before_action :enable_embedded_shopfront
 
-    # Ignores invoice orders, only order where state: 'complete'
+    # Ignores invoice orders
     def show
-      @orders = @user.orders.where(state: 'complete').order('completed_at desc')
+      @orders = orders_collection
+
+      customers = spree_current_user.customers
+      @shops = Enterprise
+        .where(id: @orders.pluck(:distributor_id).uniq | customers.pluck(:enterprise_id))
```

Comment:
I'm wondering if `distinct` could be used here instead of `uniq`, if we're going for performance? Also I think calling `uniq` on a relation is deprecated in Rails 5.

Reply:
I plan to create a separate issue for it. Stay tuned.